### DISCUSSION TITLE
Development of the Typescript extension.

### DISCUSSION THREAD

**Discussion by tsxalchemist** [foxglove/community#65] [Q&A] [answered]
Hello! I want to develop an extension for foxglove-studio using the api. I have already created a test extension, based on the documentation, got the result in foxglove (so far just a string).

During development, I'm used to using:
- Browser
- React
- TypeScript
- Sass
- Webpack

**Question**: is there any way to _live-develop_ (webpack-dev-server) the extension and see the result, and if there is, where do I read about it?

**Repository**:
https://github.com/Moloko20/gw-extension

**Comment by defunctzombie** [comment, marked as answer]
> **Question**: is there any way to _live-develop_ (webpack-dev-server) the extension and see the result, and if there is, where do I read about it?

"Somewhat". The best workflow for this is by making a Storybook story for your extension (in your repo) and mocking out the various aspects of the api that you want to simulate. That will let you iterate quicker than building and installing the extension into Studio.

@jhurliman recently did some work with Storybooks and extensions and might have an example we can make more widely available.
